CONTRIBUTING.md

```diff
@@ -0,0 +1,101 @@
+
+## Table of Contents
+
+- [Contributing to go-oscal](#contributing-to-go-oscal)
+  - [Table of Contents](#table-of-contents)
+  - [Code of Conduct](#code-of-conduct)
+  - [Getting Started](#getting-started)
+    - [Setup](#setup)
+  - [Submitting a Pull Request](#submitting-a-pull-request)
+    - [PR Requirements](#pr-requirements)
+  - [Running Tests](#running-tests)
+    - [Automated Tests](#automated-tests)
+    - [Run Tests Locally](#run-tests-locally)
+    - [Test a Local Development Version](#test-a-local-development-version)
+  - [Documentation](#documentation)
+    - [Architecture Design Records (ADR)](#architecture-design-records-adr)
+    - [How to use `adr-tools`](#how-to-use-adr-tools)
+  - [Contact](#contact)
+
+## Code of Conduct
+
+Please follow our [Code of Conduct](CODE_OF_CONDUCT.md) to maintain a respectful and collaborative environment.
+
+## Getting Started
+
+- **Repository**: [https://github.com/defenseunicorns/go-oscal/](https://github.com/defenseunicorns/go-oscal/)
+- **Go Binaries**: [https://github.com/defenseunicorns/go-oscal/releases](https://github.com/defenseunicorns/go-oscal/releases)
+- **Required Go version**: `>=1.21.5`
+
+### Setup
+
+1. Fork the repository.
+2. Clone your fork locally: `git clone https://github.com/your-username/go-oscal.git`.
+3. Create a new branch for your feature or fix: `git checkout -b my-feature-branch`.
+
+## Submitting a Pull Request
+
+1. **Create an Issue**: For significant changes, please create an issue first, describing the problem or feature proposal. Trivial fixes do not require an issue.
+2. **Commit Your Changes**: Make your changes and commit them. All commits must be signed. For help follow this [guide.](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits)
+3. **Run Tests**: Ensure that your changes pass all tests by running `go test` for each required test.
+4. **Push Your Branch**: Push your branch to your fork on GitHub.
+5. **Create a Pull Request**: Open a pull request against the `main` branch of the go-oscal repository. Please make sure that your PR passes all CI checks.
+
+### PR Requirements
+
+- PRs must be against the `main` branch.
+- PRs must pass CI checks.
+- All commits must be signed.
+- PRs should have a related issue, except for trivial fixes.
+
+## Running Tests
+
+### Automated Tests
+
+Automated tests will begin based on the paths you have edited in your Pull Request.
+⚠️ NOTE: If you are an external third-party contributor, the pipelines won't run until a [CODEOWNER](CODEOWNERS) approves the pipeline run.
+
+### Run Tests Locally
+
+- Run E2E tests: `make test`
+
+### Test a Local Development Version
+
+1. Run `make build` and wait for completion.
+2. Change to the go-oscal directory: `cd bin/go-oscal`.
+3. You can now run any of the `go-oscal` commands.
+
+## Documentation
+
+The decision for processes and docs on how-to can be found in `adr/` or in `docs/`.
+
+### Architecture Design Records (ADR)
```

Comment:
[No change required] I don't believe we have this (ADR) structure in place today. That doesn't mean we shouldn't in the future as we approach decisions that have significant impact to the project.

Reply:
I agree, I was going to pull it out, but saw the `- The criteria for when an ADR is needed is undefined. The team will decide when the team needs an ADR.` line and figured it would be good for future reference.